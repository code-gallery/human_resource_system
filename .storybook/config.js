import { configure } from '@storybook/react'

const loadStories = () => {
  require('../src/stories')
}

configure(loadStories, module)
