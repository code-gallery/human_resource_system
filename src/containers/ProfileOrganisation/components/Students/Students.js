import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { ROUTE_URL } from 'containers/constants'
import ContentCard from 'components/ContentCard'
import AvatarBig from 'components/AvatarBig'

const Students = ({ organisation, editMode }) => {
  const { students } = organisation
  const css = classNames({ 'edit-mode': editMode })

  if (students.length === 0) {
    return null
  }

  return (
    <ContentCard className={css} title="Verified Students">
      <div className="row">
        {
          students.map((item, index) => {
            const link = (item.unique_key) ?
              `${ROUTE_URL.profile}/${item.unique_key}` :
              `${ROUTE_URL.profile}/${item.id}`
            return (
              <div key={index} className="col-xs-6 col-sm-4">
                <AvatarBig
                  imageUrl={item.profile_image}
                  link={link}
                  name={`${item.first_name} ${item.last_name}`}
                  tagline={item.tagline}
                />
              </div>
            )
          })
        }
      </div>
    </ContentCard>
  )
}

Students.propTypes = {
  editMode: PropTypes.bool,
  organisation: PropTypes.shape({
    students: PropTypes.array
  })
}

export default Students
