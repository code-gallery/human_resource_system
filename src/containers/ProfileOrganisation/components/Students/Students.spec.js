import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'
import config from 'store'
import Students from './Students'

describe('containers/ProfileOrganisation/components/Students', () => {
  const { store } = config
  describe('when there are no students', () => {
    describe('when editMode is false', () => {
      it('renders correctly', () => {
        const props = {
          organisation: {
            students: []
          },
          editMode: false
        }
        const tree = renderer.create(
          <Provider store={store}>
            <Router>
              <Students {...props} />
            </Router>
          </Provider>
        ).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })

    describe('when editMode is true', () => {
      it('renders correctly', () => {
        const props = {
          organisation: {
            students: []
          },
          editMode: true
        }
        const tree = renderer.create(
          <Provider store={store}>
            <Router>
              <Students {...props} />
            </Router>
          </Provider>
        ).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })
  })

  describe('when there are some employees', () => {
    describe('when editMode is true', () => {
      it('renders correctly', () => {
        const props = {
          organisation: {
            students: [
              { unique_key: 'jane', profile_image: null, first_name: 'jane', last_name: 'doe', tagline: 'hello' },
              { id: '21', profile_image: 'https://profile_image', first_name: 'john', last_name: 'doe', tagline: 'hello' }
            ]
          },
          editMode: true
        }
        const tree = renderer.create(
          <Provider store={store}>
            <Router>
              <Students {...props} />
            </Router>
          </Provider>
        ).toJSON()

        expect(tree).toMatchSnapshot()
      })
    })
  })
})
