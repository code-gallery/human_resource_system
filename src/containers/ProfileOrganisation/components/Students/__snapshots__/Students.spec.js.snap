// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/ProfileOrganisation/components/Students when there are no students when editMode is false renders correctly 1`] = `null`;

exports[`containers/ProfileOrganisation/components/Students when there are no students when editMode is true renders correctly 1`] = `null`;

exports[`containers/ProfileOrganisation/components/Students when there are some employees when editMode is true renders correctly 1`] = `
<div
  className="ContentCard col-xs-12 edit-mode"
>
  <div
    className="ContentCard-wrapper"
  >
    <h3
      className="ContentCard-title"
    >
      Verified Students
    </h3>
    <div
      className="row"
    >
      <div
        className="col-xs-6 col-sm-4"
      >
        <div
          className="AvatarBig"
        >
          <img
            alt="jane doe"
            src="appii-placeholder.png"
          />
          <p>
            <a
              href="/profile/jane"
              onClick={[Function]}
              title="jane doe"
            >
              jane doe
            </a>
            <span
              title="hello"
            >
              hello
            </span>
          </p>
        </div>
      </div>
      <div
        className="col-xs-6 col-sm-4"
      >
        <div
          className="AvatarBig"
        >
          <img
            alt="john doe"
            src="https://profile_image"
          />
          <p>
            <a
              href="/profile/21"
              onClick={[Function]}
              title="john doe"
            >
              john doe
            </a>
            <span
              title="hello"
            >
              hello
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
